src/main.cpp

```diff
@@ -5880,6 +5886,10 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 return ProcessMessage(pfrom, NetMsgType::HEADERS, vHeadersMsg, nTimeReceived, chainparams, connman);
             }
         }
+        } // cs_main
+
+        if (fBlockReconstructed)
+            return ProcessMessage(pfrom, NetMsgType::BLOCKTXN, blockTxnMsg, nTimeReceived, chainparams, connman);
```

Comment:
Looks like cs_main is still held during the ProcessMessage call above on line 5886. Maybe deserves a comment if this is intentional.